src/components/MoneyTemporaryForRefactorRequestConfirmationList.js

```diff
@@ -614,6 +617,221 @@ function MoneyTemporaryForRefactorRequestConfirmationList({
         );
     }, [isReadOnly, iouType, selectedParticipants.length, confirm, bankAccountRoute, iouCurrencyCode, policyID, splitOrRequestOptions, formError, styles.ph1, styles.mb2, translate]);
 
+    const classifiedFields = [
+        shouldShowSmartScanFields && {
+            item: (
+                <MenuItemWithTopDescription
+                    key={translate('iou.amount')}
+                    shouldShowRightIcon={!isReadOnly && !isDistanceRequest}
+                    title={formattedAmount}
+                    description={translate('iou.amount')}
+                    interactive={!isReadOnly}
+                    onPress={() => {
+                        if (isDistanceRequest) {
+                            return;
+                        }
+                        if (isEditingSplitBill) {
+                            Navigation.navigate(ROUTES.EDIT_SPLIT_BILL.getRoute(reportID, reportActionID, CONST.EDIT_REQUEST_FIELD.AMOUNT));
+                            return;
+                        }
+                        Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_AMOUNT.getRoute(iouType, transaction.transactionID, reportID, Navigation.getActiveRouteWithoutParams()));
+                    }}
+                    style={[styles.moneyRequestMenuItem, styles.mt2]}
+                    titleStyle={styles.moneyRequestConfirmationAmount}
+                    disabled={didConfirm}
+                    brickRoadIndicator={shouldDisplayFieldError && TransactionUtils.isAmountMissing(transaction) ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : ''}
+                    error={shouldDisplayFieldError && TransactionUtils.isAmountMissing(transaction) ? translate('common.error.enterAmount') : ''}
+                />
+            ),
+            isRelevant: true,
```

Comment:
I don't like the name `isRelevant`. It makes me think that it won't be rendered if it's not relevant. I know that we're trying to determine if it should display above the "Show more" button. What about inverting the logic and going with this name instead `isSupplementary`? I think that's more clear. Or alternatively `isPrimaryField`.

Then below we can render `primaryFields` and `supplementaryFields`.

Reply:
I assumed that it is fine to tell that fields aren't much relevant if we hide them by default, but I don't have strong opinions here.

I think that I would choose flipping the logic and going with `isSupplementary` / `primaryFields` / `supplementaryFields`